### PR TITLE
Base refactor for python tests

### PR DESCRIPTION
**Pull request by Smjert**
This is the base and first part of the bigger PR (https://github.com/osquery/osquery/pull/5836) which will remain as a reference.

While this still contains a good amount of changes, it's the minimum to get the subsequent PRs merged, which add and fix the tests.
I have separated as much as possible the code in their own commit where it made sense.

Since this PR contains a change to the Docker image, I will need to adjust the Docker image name used when we decide that it's ready to be merged, which is why I've added the respective label.

(I've reopened it due to EasyCLA not kicking in. Also rebased on latest master).

### CODE REVIEW

File: libraries/cmake/facebook/modules/api.cmake [hunk 1]
```diff
@@ -199,15 +201,12 @@ function(importFacebookLibrary library_name)
...
 
 if("${THIRD_PARTY_REPOSITORY_URL}" STREQUAL "")
   message(FATAL_ERROR "The THIRD_PARTY_REPOSITORY_URL variable was not found. Has options.cmake been included?")
 endif()
 
-if(NOT PYTHON_PATH_FOLDER_INITIALIZED)
-  initializePythonPathFolder()
-  set(PYTHON_PATH_FOLDER_INITIALIZED true)
-endif()
+initializePythonPathFolder()
```
Comment (theopolis) [inline libraries/cmake/facebook/modules/api.cmake]: nitpick, no newline

**Comment by Smjert** [comment]
Added missing newlines at the end of a couple of files.
Rebased on master.

Now the only issue is EasyCLA getting stuck most likely for the Co-Authored-By.

**Comment by Smjert** [comment]
Keep forgetting to push the Docker image ^^'
Rerun the failing jobs.

**Comment by theopolis** [comment]
Looks like the docker image is still a hash instead of `v6` (guessing this is the expected change). The CLA is still having issues too.

**Comment by Smjert** [comment]
> Looks like the docker image is still a hash instead of `v6` (guessing this is the expected change). The CLA is still having issues too.

Yeah I keep that hash to avoid colliding with other PRs, that's fine; I'll change to v6 as the last step.
For the CLA I still have to report the issue.

**Comment by Smjert** [comment]
So, I've been doing some tests with a couple of PRs and `Co-Authored-By` doesn't seem to be the culprit...
I'll try to quickly close and reopen the PR, maybe something has been fixed.

**Comment by theopolis** [comment]
We can also disable the GitHub requirement for the EasyCLA test until it’s fixed. And add the “cla signed” tag after a manual lookup.

**Comment by Smjert** [comment]
> We can also disable the GitHub requirement for the EasyCLA test until it’s fixed. And add the “cla signed” tag after a manual lookup.

That's fine by me, I was opening a Jira ticket pointing to this PR, but if we're going to merge it I'll avoid it, especially since I've done a couple of tests with `Co-Authored-By` and the same Co-Author and it doesn't seem to get stuck.
There's something strange with this specific set of commits.

So I'm going to put the correct version for the Docker image.

**Comment by theopolis** [comment]
Ready to rebase-merge?

**Comment by Smjert** [comment]
> Ready to rebase-merge?

Yeah! Just need someone to cover EasyCLA eyes!

**Comment by theopolis** [comment]
Sorry that we let this sit and now it needs a rebase. :/

**Comment by inferno-chromium** [comment]
looks the configs/ dir move broke OSS-Fuzz integration, can you please fix the scripts, see failure in https://oss-fuzz-build-logs.storage.googleapis.com/index.html#osquery

**Comment by Smjert** [comment]
> looks the configs/ dir move broke OSS-Fuzz integration, can you please fix the scripts, see failure in https://oss-fuzz-build-logs.storage.googleapis.com/index.html#osquery

Thank you for the heads up!

**Pull request merged**